src/oca_github_bot/tasks/migration_issue_bot.py

```diff
@@ -0,0 +1,103 @@
+# Copyright 2019 Tecnativa - Pedro M. Baeza
+# Distributed under the MIT License (http://opensource.org/licenses/MIT).
+
+import re
+
+from .. import github
+from ..config import switchable
+from ..process import check_call
+from ..queue import getLogger, task
+from .merge_bot import _user_can_merge
+
+_logger = getLogger(__name__)
+
+
+def _create_or_find_branch_milestone(repo, branch):
+    for milestone in repo.milestones():
+        if milestone.title == branch:
+            return milestone
+    return repo.create_milestone(branch)
+
+
+@task()
+@switchable("migration_issue_bot")
+def migration_issue_start(org, repo, pr, username, module=None, dry_run=False):
+    with github.login() as gh:
+        gh_pr = gh.pull_request(org, repo, pr)
+        target_branch = gh_pr.base.ref
+        pr_branch = f"tmp-pr-{pr}"
+        try:
+            with github.temporary_clone(org, repo, target_branch) as clone_dir:
+                # Create merge bot branch from PR and rebase it on target branch
+                # This only serves for checking permissions
+                check_call(
+                    ["git", "fetch", "origin", f"pull/{pr}/head:{pr_branch}"],
+                    cwd=clone_dir,
+                )
+                check_call(["git", "checkout", pr_branch], cwd=clone_dir)
+                if not _user_can_merge(
+                    gh, org, repo, username, clone_dir, target_branch
+                ):
+                    github.gh_call(
+                        gh_pr.create_comment,
+                        f"Sorry @{username} you are not allowed to mark the addon to"
+                        f"be migrated.\n\n"
+                        f"To do so you must either have push permissions on "
+                        f"the repository, or be a declared maintainer of all "
+                        f"modified addons.\n\n"
+                        f"If you wish to adopt an addon and become it's "
+                        f"[maintainer]"
+                        f"(https://odoo-community.org/page/maintainer-role), "
+                        f"open a pull request to add "
+                        f"your GitHub login to the `maintainers` key of its "
+                        f"manifest.",
+                    )
```

Comment:
Is there a persmission concern here? I mean anyone can create a migration PR and it can be added to the list automatically. Actually we could even look at the list of new modules in the PR (with a variant of `git_modified_addons` that returns new addons only), then find the migration issue and link/add modules to it.

Reply:
Yes, there is. Not everyone must be able to do it, because people can hijack otherwise valid migration PRs with new ones, so PSCs/maintainers must decide, and not any other.

Reply:
I agree about the PR hijack, but why not letting any user add a module or link it with a PR if none is set ?

Reply:
I would agree to let the maintainers do it, but not anyone. See what has happened with your confusion about the split of OCA/hr into OCA/hr, OCA/hr-holidays, OCA/hr-expense, etc. It can become very contaminated. And having this tool, it will be very easy to maintainers to raise the message to do it (or even me, that I'm doing it manually for a lot of them). Anyways, I only do it when CI is green and a few rules are respected (like preserving commit history), and that's why again I don't think this should be done by anyone.